packages/local_auth/local_auth/windows/local_auth_plugin.cpp

```diff
@@ -0,0 +1,187 @@
+
+  registrar->AddPlugin(std::move(plugin));
+}
+
+LocalAuthPlugin::LocalAuthPlugin(flutter::PluginRegistrarWindows* registrar)
+    : registrar_(registrar) {}
+
+LocalAuthPlugin::~LocalAuthPlugin() {}
+
+void LocalAuthPlugin::HandleMethodCall(
+    const MethodCall<EncodableValue>& method_call,
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  if (method_call.method_name().compare("authenticate") == 0) {
+    Authenticate(method_call, std::move(result));
+  } else if (method_call.method_name().compare("getAvailableBiometrics") == 0) {
+    GetAvailableBiometrics(std::move(result));
+  } else if (method_call.method_name().compare("isDeviceSupported") == 0) {
+    IsDeviceSupported(std::move(result));
+  } else {
+    result->NotImplemented();
+  }
+}
+
+winrt::fire_and_forget LocalAuthPlugin::Authenticate(
+    const MethodCall<EncodableValue>& method_call,
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  auto reasonW = s2ws(GetArgument<std::string>(
+      "localizedReason", method_call.arguments(), std::string()));
+
+  auto ucvAvailability = co_await Windows::Security::Credentials::UI::
+      UserConsentVerifier::CheckAvailabilityAsync();
+  if (ucvAvailability != Windows::Security::Credentials::UI::
+                             UserConsentVerifierAvailability::Available) {
+    result->Error("NotAvailable", "Required security features not enabled");
+    co_return;
+  }
+
+  auto userConsentVerifierInterop = winrt::get_activation_factory<
+      Windows::Security::Credentials::UI::UserConsentVerifier,
+      IUserConsentVerifierInterop>();
+
+  auto hWnd = GetRootWindow(registrar_->GetView());
+
+  HSTRING hReason;
+  WindowsCreateString(reasonW.c_str(), (uint32_t)reasonW.size(), &hReason);
+
+  try {
+    auto consentResult =
+        co_await winrt::capture<Windows::Foundation::IAsyncOperation<
+            Windows::Security::Credentials::UI::UserConsentVerificationResult>>(
+            userConsentVerifierInterop,
+            &::IUserConsentVerifierInterop::RequestVerificationForWindowAsync,
+            hWnd, hReason);
+
+    result->Success(EncodableValue(
+        consentResult == Windows::Security::Credentials::UI::
+                             UserConsentVerificationResult::Verified));
+  } catch (...) {
+    result->Success(EncodableValue(false));
+  }
+  WindowsDeleteString(hReason);
+}
+
+winrt::fire_and_forget LocalAuthPlugin::GetAvailableBiometrics(
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  try {
+    flutter::EncodableList biometrics;
+    auto ucvAvailability = co_await Windows::Security::Credentials::UI::
+        UserConsentVerifier::CheckAvailabilityAsync();
+    if (ucvAvailability == Windows::Security::Credentials::UI::
+                               UserConsentVerifierAvailability::Available) {
+      biometrics.push_back(EncodableValue("fingerprint"));
+      biometrics.push_back(EncodableValue("face"));
+      biometrics.push_back(EncodableValue("iris"));
```

Comment:
This list is clearly not accurate. What actual information do we get about supported authentication options? We're in the process of changing the API (as part of the federation PRs) to also list `strong` and `weak` to match the Android options. Does one or both of those work instead? Is there a different set of things we should consider adding?

Reply:
The existing Windows API being used here does not let us ask for a specific auth method, so it is up to the user to choose one that is available on their machine. That can be a combination of Fingerprint, Face, and or PIN.
I will ask internally here at Microsoft if there is guidance on querying those methods, but I'm assuming it would require some complex device GUID querying work, with likely false positives/negatives, but again, just an assumption.

Reply:
If there's any reasonable correspondence with either the "weak" or "strong" designations that Android uses now we could just return that.

If we can't get *any* information would be to add something like "unknown" to the set of possible options in the API. Then clients can decide how they want to handle it.

Reply:
I've started an internal thread on how this should be handled, and I'll get back to you with the recommendations, but I believe this is either gonna be a new value that needs to be created only for windows or, for better compatibility, 'weak'.